js/example_roulette.js

```diff
@@ -0,0 +1,30 @@
+/**
+Picking the frontpage example after the page has loaded leads to reflows
+and re-rendering. Hence, we pick the example as soon as possible
+
+Copyright: 1999-2017 by D Language Foundation
+License:   http://boost.org/LICENSE_1_0.txt, Boost License 1.0
+*/
+/**
+Searches any object for the first hit.
+Params:
+    el = object to filter
+    fn = function to be applied as filter
+Returns: first hit, `undefined` otherwise
+*/
+function findFirst(el, fn) {
+    return Array.prototype.filter.call(el, fn)[0];
+}
+(function() {
+// randomly pick one example and bootstrap the runnable editor
+var examples = document.getElementsByClassName('your-code-here-extra');
+var rouletteIndex = Math.floor(Math.random() * examples.length);
+var rouletteChild = examples[rouletteIndex];
+
+// step 1: select the .runnable-example <div>
+var rouletteChildNode = findFirst(rouletteChild.children, function(e) { return e.className.indexOf("runnable-example") != -1; });
+// step 2: select the first <pre> element (of the .runnable-example <div>)
+var el = findFirst(rouletteChildNode.children, function(e) { return e.nodeName == "PRE";});
+el.style.marginBottom = "20px";
```

Comment:
Oh, I understand what you mean now. Just because you need to reset it doesn't mean you have to add it via JS too - JS styles override CSS ones because they're going to be set on the element itself. Regardless I agree that undoing/resetting it is suboptimal, see my comment below.